### PR TITLE
converted to es6

### PR DESCRIPTION
**Pull request by orthagonal**
removed grunt stuff
updated package.json
updated tests and code to standardized es6
added travisci  support

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -1,10 +0,0 @@
-# editorconfig.org
```
Comment (jgallen23) [inline .editorconfig]: this file should stay

File: lib/parse-dir.js [hunk 2]
```diff
@@ -1,30 +1,30 @@
...
-  var jsonParser = require('./parsers/json');
-  var yamlParser = require('./parsers/yaml');
-  var jsParser = require('./parsers/javascript');
+  const jsonParser = require('./parsers/json');
+  const yamlParser = require('./parsers/yaml');
+  const jsParser = require('./parsers/javascript');
 
-  var ParseDir = function() {
-    var that = this;
+  const ParseDir = function() {
```
Comment (jgallen23) [inline lib/parse-dir.js]: we should change this to a `class`

File: lib/parse-dir.js [hunk 3]
```diff
@@ -64,20 +60,20 @@
...
+    const meta = {
       relativePath: file,
       filepath: path.resolve(file),
       filename: path.basename(file),
       basename: path.basename(file, path.extname(file)),
       extension: path.extname(file),
       parsed: false
     };
-    var that = this;
+    const that = this;
```
Comment (jgallen23) [inline lib/parse-dir.js]: shouldn't need this anymore with arrow functions

File: lib/parse-dir.js [hunk 4]
```diff
@@ -144,8 +140,7 @@
             jsParser(meta, done);
             break;
           default:
-            done();
-            break;
+            return done();
         }
       }.bind(this)], function(err, data) {
```
Comment (jgallen23) [inline lib/parse-dir.js]: can get rid of bind

File: lib/parsers/javascript.js [hunk 5]
```diff
@@ -1,16 +1,16 @@
+// registers coffeescript with 'require'
+require('coffee-script/register');
```
Comment (jgallen23) [inline lib/parsers/javascript.js]: what is this for?

File: package.json [hunk 6]
```diff
@@ -18,18 +18,19 @@
...
+    "eslint-config-firstandthird": "^2.0.0",
+    "eslint-plugin-import": "^1.5.0",
+    "eslint-plugin-jsx-a11y": "^1.0.2",
+    "eslint-plugin-react": "^5.1.1",
+    "mocha": "^2.5.3"
   },
   "dependencies": {
-    "glob": "^4.0.0",
     "async": "^0.9.0",
+    "coffee-script": "^1.10.0",
```
Comment (jgallen23) [inline package.json]: why was this added?

**Comment by Belelros** [comment]
Same as @jgallen23 . Not sure I see the coffee-script being used and it's not something we're using anywhere. Removing `that` and `bind`s should be also good. Also eager to see the class. Will re-review once that's done

File: package.json (hunk 6 above)
Comment (orthagonal) [inline package.json]: That dependency upgrades the 'require' keyword, so that node can require coffeescript modules just like they were js modules.  Removing it breaks parse-dir's support for coffeescript.

Comment (dawnerd) [inline package.json]: Why would we need coffee script?

Comment (jgallen23) [inline package.json]: It was working before if coffee script was loaded by the parent project.

File: lib/parse-dir.js [hunk 7]
```diff
@@ -1,168 +1,183 @@
...
+const jsParser = require('./parsers/javascript');
+
+// used by ParseDir to process and parse individual files:
+class FileProcessor {
+  constructor(options) {
+    this.data = [];
+    this.__async = options.async !== undefined ? options.aync : true;
+  }
+
+  __determineIfDirectory (done, results) {
```
Comment (Belelros) [inline lib/parse-dir.js]: Not sure about the `__`+ name pattern is a thing on JS or our libs. Reminds me to Python. Usually private things are with a single underscore at the end of the name. `determineIfDirectory_`.

@jgallen23 thoughts?

File: lib/parse-dir.js [hunk 8]
```diff
@@ -1,168 +1,183 @@
...
-      basename: path.basename(file, path.extname(file)),
-      extension: path.extname(file),
-      parsed: false
+    if (results.__async) {
+      fs.lstat(results.meta.filepath, onLsStat);
+    } else {
+      onLsStat(null, fs.lstatSync(results.meta.filepath));
+    }
+  }
+  __getFileContents (done, results) {
```
Comment (Belelros) [inline lib/parse-dir.js]: We should strive for consistency. Constructor gets like this: `constructor()` (no space after function name) but other functions do get the space. I think we've been using no space. 

@jgallen23 ?

File: lib/parse-dir.js [hunk 9]
```diff
@@ -1,168 +1,183 @@
...
+    }
+  }
+
+  processFile (file, callback) {
+    const isAsync = this.__async;
+
+    async.auto({
+      __async: (done) => {
+        return done(null, isAsync);
       },
```
Comment (Belelros) [inline lib/parse-dir.js]: We could leverage the auto return here and just below.

`__async: done => done(null, isAsync),`

File: lib/parse-dir.js [hunk 10]
```diff
@@ -1,168 +1,183 @@
...
+}
+
+// front-end to parse a directory at a time:
+class ParseDir {
+  constructor(options) {
+    this.__async = options.async !== undefined ? options.aync : true;
+    this.fileParser = new FileProcessor(options);
+  }
+
+  sync (pattern) {
```
Comment (Belelros) [inline lib/parse-dir.js]: I think we should normalise spacing between methods too. This class gets spaces between methods but not the `FileProcessor` class. My preference is to use more new lines in general as things feel less packed for me. 

@jgallen23 thoughts on this?

File: lib/parse-dir.js (hunk 7 above)
Comment (orthagonal) [inline lib/parse-dir.js]: I'd actually prefer no underscores at all here.  I'm cool with whatever you guys prefer though.

File: lib/parse-dir.js (hunk 8 above)
Comment (orthagonal) [inline lib/parse-dir.js]: I find the extra space makes it easier to scan the parameter lists when you collapse the class down in my editor.  Either way, you're right the constructor should look the same as the other methods.

File: lib/parse-dir.js (hunk 10 above)
Comment (orthagonal) [inline lib/parse-dir.js]: 1 line spacing and 1 space between method name and param list makes it really easy to read when you collapse it.

File: lib/parse-dir.js (hunk 7 above)
Comment (jgallen23) [inline lib/parse-dir.js]: yeah, no underscores

File: package.json [hunk 11]
```diff
@@ -18,18 +18,16 @@
...
+    "chai": "^3.5.0",
+    "coffee-script": "^1.10.0",
+    "eslint": "^3.9.1",
+    "eslint-config-firstandthird": "^3.0.2",
+    "eslint-plugin-import": "^2.1.0",
+    "mocha": "^2.5.3"
   },
   "dependencies": {
-    "glob": "^4.0.0",
     "async": "^0.9.0",
```
Comment (jgallen23) [inline package.json]: we should update async

File: package.json [hunk 12]
```diff
@@ -18,18 +18,16 @@
...
-    "grunt-contrib-jshint": "^0.10.0",
-    "grunt-contrib-watch": "^0.6.1",
-    "grunt-mocha-test": "^0.10.2",
-    "load-grunt-config": "^0.9.2"
+    "chai": "^3.5.0",
+    "coffee-script": "^1.10.0",
+    "eslint": "^3.9.1",
+    "eslint-config-firstandthird": "^3.0.2",
+    "eslint-plugin-import": "^2.1.0",
+    "mocha": "^2.5.3"
```
Comment (jgallen23) [inline package.json]: can we switch to tap / tap-spec?

Comment (jgallen23) [inline package.json]: actually, hold on this, we'll do that in a separate PR
